python/pyarrow/scalar.pxi

```diff
@@ -1084,6 +1084,10 @@ cdef class FixedShapeTensorScalar(ExtensionScalar):
             ctensor = GetResultValue(c_type.MakeTensor(scalar))
         return pyarrow_wrap_tensor(ctensor)
 
+cdef class Bool8Scalar(ExtensionScalar):
+    """
+    Concrete class for bool8 extension scalar.
+    """
```

Comment:
We could probably override the `as_py()` method here, which by default uses the storage value. But in this case we could convert that integer to a bool?

Reply:
Implemented and added tests